src/fides/api/ctl/migrations/versions/5307999c0dac_remove_deprecated_data_uses_for_.py

```diff
@@ -0,0 +1,32 @@
+"""remove deprecated data uses for fideslang 1.4
+
+Revision ID: 5307999c0dac
+Revises: 76c02f99eec1
+Create Date: 2023-06-11 11:15:53.386526
+
+"""
+from alembic import op
+
+from loguru import logger
+from sqlalchemy import text
+from sqlalchemy.engine import Connection
+
+
+# revision identifiers, used by Alembic.
+revision = "5307999c0dac"
+down_revision = "76c02f99eec1"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    logger.info("Removing default data uses.")
+    bind: Connection = op.get_bind()
+    bind.execute(text("DELETE FROM ctl_data_uses WHERE is_default = TRUE;"))
```

Comment:
Are you sure you really want to delete _all_ the data use table and then recreate them afterwards, instead of attempting a selective delete? In this file you have an explicit list of keys that you attempt to rename: https://github.com/ethyca/fides/blob/0d721d1b2e02653406d50ea5ef071024e6789d4f/src/fides/api/ctl/migrations/versions/63b482f5b49b_upgrade_to_fideslang_1_4.py#L24

Reply:
In other words: you're sure that both:
1. There aren't any foreign-key relationships that might cause a SQL error here on delete
2. All the unaffected data uses will safely be recreated afterwards
3. Nothing else is risky here?

Reply:
guaranteed 🙂 check the `steps to confirm` up above

We have no foreign key relationships at all to `data use`, all validation is done via the API which doesn't get touched until after the migrations/seeding is done. I can write up more test cases to confirm this

Correct that in that other migration we _are_ doing selective updates, but that's because we're touching tables outside of `data use`. Here we're only deleting rows that we know we added ourselves (default taxonomy) and that will get re-populated correctly as soon as the migrations finish.

No, but again our test coverage of version-over-version is not broad, so we should do more manual testing to verify. I can work on more test cases to be a little safer?

Reply:
OK, sounds good. Let's ship it then, and it feels like the only risk might be on the `CustomField` model which is pretty manageable?

Reply:
Yep, we'll do some special test cases there and I'll follow up with the relevant individuals 🙂